tools/api_proto_breaking_change_detector/detector.py

```diff
@@ -0,0 +1,223 @@
+        :type path_to_before: ``str``
+
+        :param path_to_after: absolute path to the .proto file in the after state
+        :type path_to_after: ``str``
+
+        :param additional_args: additional arguments for the breaking change detector CLI. May be tool-dependent.
+            Additional arguments are passed in both the "initialize" and "detect changes" steps.
+        :type additional_args: ``List[str]``
+        """
+        if not Path(path_to_before).is_file():
+            raise ValueError(f"path_to_before {path_to_before} does not exist")
+
+        if not Path(path_to_after).is_file():
+            raise ValueError(f"path_to_after {path_to_after} does not exist")
+
+        self._path_to_before = path_to_before
+        self._path_to_after = path_to_after
+        self._additional_args = additional_args
+
+    def run_detector(self) -> None:
+        """Runs the breaking change detector with the parameters given in ``__init__``. This method should populate
+        the detector's internal data such that `is_breaking` and `lock_file_changed` do not require any additional
+        invocations to the breaking change detector.
+        """
+        # 1) pull buf BSR dependencies with buf mod update (? still need to figure this out. commented out for now)
+        # 2) copy buf.yaml into temp dir
+        # 3) copy start file into temp dir
+        # 4) buf build -o tmp_file
+        # 5) copy changed file into temp dir
+        # 6) buf breaking --against tmp_file
+        # 7) check for differences (if changes are breaking, there should be none)
+
+        with tempfile.TemporaryDirectory(
+                prefix=str(Path(".").absolute())
+                + os.sep) as temp_dir:  # buf requires protobuf files to be in a subdirectory of cwd
+            buf_path = runfiles.Create().Rlocation("com_github_bufbuild_buf/bin/buf")
+
+            buf_config_loc = Path(".", "tools", "api_proto_breaking_change_detector")
+
+            #copyfile(Path(buf_config_loc, "buf.lock"), Path(".", "buf.lock")) # not needed? refer to comment below
+            yaml_file_loc = Path(".", "buf.yaml")
+            copyfile(Path(buf_config_loc, "buf.yaml"), yaml_file_loc)
+
+            # TODO: figure out how to automatically pull buf deps
+            # `buf mod update` doesn't seem to do anything, and the first test will fail because it forces buf to automatically start downloading the deps
+            #        bcode, bout, berr = run_command(f"{buf_path} mod update")
+            #        bout, berr = ''.join(bout), ''.join(berr)
+            target = Path(temp_dir, f"{Path(self._path_to_before).stem}.proto")
+
+            buf_args = [
+                "--path",
+                str(target.relative_to(
+                    Path(".").absolute())),  # buf requires relative pathing for roots...
+                "--config",
+                str(yaml_file_loc),
+            ]
+            if self._additional_args is not None:
```

Comment:
You can avoid this extra conditional by making sure `self._additional_args` is an empty list if not provided, or by using Python's truthiness rules to replace `self._additional_args` on the next line with `self._additional_args or []`. Since `None` and `[]` are both false-y, the result of the expression would be the empty list.